src/pages/Groups.tsx

```diff
@@ -28,7 +29,8 @@ export const Groups: MainStackScreen<"Groups"> = () => {
 
   const [filters, setFilters] = useState<Filters>({})
 
-  const [groups, setGroups] = useState<Group[]>([])
+  const [grps] = useApiCall(api.groups.getFromGithub, {}, [])
+  const groups = grps ?? []
 
   const [filteredGroups, setFilteredGroups] = useState<Group[]>([])
```

Comment:
Volevo evidenziare il punto sotto ma github non me lo fa fare perchè non hai toccato quelle linee. Comunque nello useEffect dei filtri penso vada cambiata la dependency `groups ` in `grps`. Al momento lo useEffect sta andando in loop infinito probabilmente perchè ogni render cambia il puntatore di `groups`.


Quindi lo useEffect dovrebbe diventare cosi:

`  //Apply filters
  useEffect(() => {
    if (isMounted && groups) {
      const newGroups = applyFilters(groups, filters)
      setFilteredGroups(newGroups)
    }
  }, [filters, grps])`

Reply:
shit non so come mi era sfuggita questa ahah